File: src/leetcode/java/p1768/Solution.java
package leetcode.java.p1768;
// https://leetcode.com/problems/merge-strings-alternately/

/**
 * Ad hoc
 * | Time: O (n + m)
 * | Space: O (n + m)
 */
public class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int l = 0, r = 0;
        while (l < word1.length() && r < word2.length()) {
            sb.append(word1.charAt(l++));
            sb.append(word2.charAt(r++));
        }
        while (l < word1.length()) {
            sb.append(word1.charAt(l++));
        }
        while (r < word2.length()) {
            sb.append(word2.charAt(r++));
        }
        return sb.toString();
    }
}

File: src/leetcode/python/p1768/solution.py
"""
https://leetcode.com/problems/merge-strings-alternately/?envType=study-plan-v2&envId=leetcode-75
"""
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i= 0
        j= 0
        ret= ""
        while i < len(word1) or j < len(word2):
            if i < len(word1):
                ret= ret + word1[i]
                i= i+1
            if j < len(word2):
                ret= ret + word2[j]
                j= j+1
        return ret